File: buzzvil-sdk-ios/buzzvil-sdk-ios-objc/Native/Carousel/CarouselFeedEntryView.m
//##ARTHUR
//#import "CarouselFeedEntryView.h"
//
//@interface CarouselFeedEntryView ()
//
//@property (nonatomic, strong, readonly) UIButton *button;
//
//@end
//
//@implementation CarouselFeedEntryView
//
//
//- (instancetype)initWithFrame:(CGRect)frame {
//  if (self = [super initWithFrame:frame]) {
//    [self setupView];
//    [self setupLayout];
//  }
//  return self;
//}
//
//- (void)setupView {
//  _button = [[UIButton alloc] initWithFrame:CGRectZero];
//  [_button setTitle:@"포인트 더 받으러 가기" forState:UIControlStateNormal];
//  [_button setTitleColor:[UIColor systemBlueColor] forState:UIControlStateNormal];
//  [self addSubview:_button];
//  self.clickableViews = @[self.button];
//}
//
//- (void)setupLayout {
//  _button.translatesAutoresizingMaskIntoConstraints = NO;
//  [NSLayoutConstraint activateConstraints:@[
//    [_button.topAnchor constraintEqualToAnchor:self.topAnchor],
//    [_button.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
//    [_button.trailingAnchor constraintEqualToAnchor:self.trailingAnchor],
//    [_button.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
//  ]];
//}
//
//@end

File: archive/buzzvil-sdk-v5-ios/buzzvil-sdk-ios-swiftui/Native/Carousel/CarouselFeedEntryView.swift
import UIKit
import BuzzvilSDK

final class CarouselFeedEntryView: BZVFeedEntryView {
  private lazy var button: UIButton = {
    let button = UIButton(type: .system)
    button.setTitle("포인트 더 받으러 가기", for: .normal)
    button.tintColor = .systemBlue
    return button
  }()
  
  override init(frame: CGRect) {
    super.init(frame: frame)
    setupView()
    setupLayout()
  }
  
  required init?(coder: NSCoder) {
    fatalError()
  }
  
  private func setupView() {
    addSubview(button)
    clickableViews = [button]
  }
  
  private func setupLayout() {
    button.translatesAutoresizingMaskIntoConstraints = false
    NSLayoutConstraint.activate([
      button.topAnchor.constraint(equalTo: topAnchor),
      button.leadingAnchor.constraint(equalTo: leadingAnchor),
      button.trailingAnchor.constraint(equalTo: trailingAnchor),
      button.bottomAnchor.constraint(equalTo: bottomAnchor),
    ])
  }
}
